src/NorthwindWebsite/Business/CustomValidators/AllowedImageFileTypesAttribute.cs

```diff
@@ -0,0 +1,44 @@
+﻿using NorthwindWebsite.Core.ApplicationSettings;
+using NorthwindWebsite.Core.Utils;
+using System.ComponentModel.DataAnnotations;
+
+namespace NorthwindWebsite.Business.CustomValidators;
+
+public class AllowedImageFileTypesAttribute : ValidationAttribute
+{
+    private string[] permittedExtentions;
+
+    public string GetErrorMessage() =>
+        string.Format("Wrong file format. Please use {0}", GetFileTypesRepresentation());
+
+    private string GetFileTypesRepresentation()
+    {
+        var joinedExtensions = string.Empty;
+
+        foreach (var extension in permittedExtentions)
+        {
+            joinedExtensions = string.Join(", ", permittedExtentions);
+        }
+
+        return joinedExtensions;
+    }
+
+    protected override ValidationResult? IsValid(
+        object? value, ValidationContext validationContext)
+    {
+        var appSettings = validationContext.GetService<AppSettings>();
+
+        permittedExtentions = appSettings.FileUploadOptions.ImageFileFormats;
+
+        var contentType = value is IFormFile uploadedFile
+            ? uploadedFile.GetContentType()
+            : string.Empty;
```

Comment:
Well, in this case I would advise to return an error because it means that you are working not with file (implementation of `IFormFile`) but with something else, and this validator is working exactly with files. So, it may return an error because current object is not a file and validation cannot be proceeded.